src/proxy-good-queue.js

```diff
@@ -0,0 +1,508 @@
+        try {
+          connection.ws.send(coinHiveMessage);
+          log("[POOL][MINER]", coinHiveMessage);
+        } catch (e) {
+          log("socket seems to be already closed.");
+          killConnection(connection);
+        }
+    } else {
+        log("failed to send message to miner cos it was offline:", coinHiveMessage);
+    }
+   }
+}
+
+//function geRPVid
+function getRpcId(connection) {
+	
+	if (typeof addressConections[connection.address].rpcIdtoUser == 'undefined' || addressConections[connection.address].rpcIdtoUser == null) {  
+		killConnection(connection);
+	}
+
+    //set new rpcId
+    var rpcId = addressConections[connection.address].rpcId++;
+	
+    //assoc rpcId to User
+    addressConections[connection.address].rpcIdtoUser[rpcId] = connection.id_user;
+    
+    
+    return rpcId;
+}
+
+//function getHashes
+function getHashes(connection) {
+  return connection.hashes++; 
+}
+ 
+//function connect new socket
+function connectSocket(connection, address) {  
+    
+    //Checking exists Socket Address
+    if (typeof addressConections[address] != 'undefined' && addressConections[address] != null) {  
+              
+        //start queue
+        connection.online = true;
+        connection.hashes = 1;
+        
+    } else {
+      
+        //Create a new socket
+        addressConections[address] = {};
+        addressConections[address].socket = new net.Socket();
+        log("[SERVER][ Socket TCP Created at", address);
+      
+        //SET UT8
+        addressConections[address].socket.setEncoding("utf8");
+        
+        //SET Buffer
+        addressConections[address].buffer = "";
+        addressConections[address].jobs = [];
+        addressConections[address].workerId = [];
+     
+		//add users online
+        addressConections[address].users = 0;
+            
+        //set rpcId, assoc to user and add 
+        addressConections[address].rpcIdtoUser = [];
+        addressConections[address].rpcIdAuths = [];
+        addressConections[address].rpcId = 1;
+	
+        //create queue to address
+        addressConections[address].queue = new Queue();
+        
+        //Queue OnClose
+        addressConections[address].queue.on("close", () => {
+            killConnection(connection);
+            log("miner connection closed");
+        });
+    
+    
+        //Queue OnError
+        addressConections[address].queue.on("error", error => {
+            killConnection(connection);
+            log("miner connection error", error.message);
+        });
+    
+        //Queue On message
+        addressConections[address].queue.on("message", function(message) {
+			
+            //set data
+            let data = null;
+        
+            //try parse json
+            try {
+                data = JSON.parse(message);
+            } catch (e) {
+                return log("can't parse message as JSON from miner:", message);
+            }
+			
+			//extract user
+			var id_user  = data.id_user;
```

Comment:
Avoid using a `var`

Reply:
¿data return a id_user?

Reply:
No, use `let` and `const` instead of `var`.